buildSrc/src/main/kotlin/profiler.android-studio-setup.gradle.kts

```diff
@@ -0,0 +1,71 @@
+import extensions.AndroidStudioTestExtension
+
+repositories {
+    ivy {
+        // Url of Android Studio archive
+        url = uri("https://redirector.gvt1.com/edgedl/android/studio/ide-zips")
+        patternLayout {
+            artifact("[revision]/[artifact]-[revision]-[ext]")
+        }
+        metadataSources { artifact() }
+    }
+}
+
+val os = System.getProperty("os.name").toLowerCase()
+val architecture = System.getProperty("os.arch").toLowerCase()
+fun isWindows(): Boolean = os.startsWith("windows")
+fun isMacOS(): Boolean = os.startsWith("mac")
+fun isLinux(): Boolean = os.startsWith("linux")
+fun isIntel(): Boolean = architecture == "x86_64" || architecture == "x86"
+fun File.isTar(): Boolean = name.endsWith(".tar.gz")
+
+val extension = extensions.create<AndroidStudioTestExtension>("androidStudioTests")
+
+val androidStudioRuntime by configurations.creating
+val autoDownloadAndroidStudio by lazy { extension.autoDownloadAndroidStudio.getOrElse(false) }
+afterEvaluate {
```

Comment:
Can we remove this `afterEvaluate` somehow? You could always add the dependency, with a lazy version I suppose. And then skip the task if you shouldn't download it.